src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -344,7 +344,12 @@ public void filterLateSubmissions(Repository repository, ProgrammingExercise exe
 
             git.close();
 
-            reset(repository, latestCommitBeforeDeadline.getId().getName());
+            if (latestCommitBeforeDeadline == null) {
+                reset(repository, "origin/master");
```

Comment:
Couldn't the latest commit on origin/master be a commit that was made after the deadline passed?
If I'm not mistaken the method should properly throw an exception in the case that there is no relevant commit, otherwise the output from the two scenarios (there is vs. there isn't a commit within the deadline) would be very different.

Reply:
Thx for the review and pointing out the actual problem. With JGit a starting date has to be set, which was set to the release date. If no commits between the release date and due date are made, an error occurs. The starting date is now hardcoded to EPOCH (1970).